File: gpversionchecker/src/main/java/com/robohorse/gpversionchecker/provider/SharedDataProvider.java
package com.robohorse.gpversionchecker.provider;

import android.content.SharedPreferences;

import com.robohorse.gpversionchecker.domain.Version;
import com.robohorse.gpversionchecker.utils.DateFormatUtils;

/**
 * Created by robohorse on 06.03.16.
 */
public class SharedDataProvider {
    private static final String GPVCH_TIME = "gpvch_time";
    private static final String GPVCH_VERSION = "gpvch_version";

    private final SharedPreferences sharedPreferences;
    private final DateFormatUtils formatUtils;

    public SharedDataProvider(SharedPreferences sharedPreferences, DateFormatUtils formatUtils) {
        this.sharedPreferences = sharedPreferences;
        this.formatUtils = formatUtils;
    }

    public void saveCurrentDate() {
        sharedPreferences
                .edit()
                .putLong(GPVCH_TIME, formatUtils.formatTodayDate().getTime())
                .apply();
    }

    public long provideLastCheckTime() {
        return sharedPreferences.getLong(GPVCH_TIME, 0L);
    }

    public void saveCurrentVersion(Version version) {
        sharedPreferences
                .edit()
                .putString(GPVCH_VERSION, version.getNewVersionCode())
                .apply();
    }

    public String provideLastVersionCode() {
        return sharedPreferences.getString(GPVCH_VERSION, null);
    }
}

File: gpversionchecker/src/main/kotlin/com/robohorse/gpversionchecker/provider/SharedDataProvider.kt
package com.robohorse.gpversionchecker.provider

import android.content.SharedPreferences
import com.robohorse.gpversionchecker.domain.Version
import com.robohorse.gpversionchecker.utils.DateFormatUtils

class SharedDataProvider(
        private val sharedPreferences: SharedPreferences,
        private val formatUtils: DateFormatUtils
) {

    fun saveCurrentDate() {
        formatUtils.formatTodayDate()?.let {
            sharedPreferences
                    .edit()
                    .putLong(GPVCH_TIME, it.time)
                    .apply()
        }
    }

    fun provideLastCheckTime(): Long {
        return sharedPreferences.getLong(GPVCH_TIME, 0L)
    }

    fun saveCurrentVersion(version: Version) {
        sharedPreferences
                .edit()
                .putString(GPVCH_VERSION, version.newVersionCode)
                .apply()
    }

    fun provideLastVersionCode(): String? {
        return sharedPreferences.getString(GPVCH_VERSION, null)
    }
}

private const val GPVCH_TIME = "gpvch_time"
private const val GPVCH_VERSION = "gpvch_version"
